AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Task/AWSAuthFetchSessionTask.swift

```diff
@@ -0,0 +1,51 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+import Amplify
+
+class AWSAuthFetchSessionTask: AuthFetchSessionTask {
+    private let request: AuthFetchSessionRequest
+    private let authStateMachine: AuthStateMachine
+    private let fetchAuthSessionHelper: FetchAuthSessionOperationHelper
+    private var stateMachineToken: AuthStateMachineToken?
+    
+    var eventName: HubPayloadEventName {
+        HubPayload.EventName.Auth.fetchSessionAPI
+    }
+
+    init(_ request: AuthFetchSessionRequest, authStateMachine: AuthStateMachine) {
+        self.request = request
+        self.authStateMachine = authStateMachine
+        self.fetchAuthSessionHelper = FetchAuthSessionOperationHelper()
+    }
+
+    func execute() async throws -> AuthSession {
+        await didConfigure()
+        return try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<AuthSession, Error>) in
+            let doesNeedForceRefresh = request.options.forceRefresh
+            fetchAuthSessionHelper.fetch( authStateMachine, forceRefresh: doesNeedForceRefresh) {result in
+                do {
+                    let session = try result.get()
+                    continuation.resume(returning: session)
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+        }
+    }
+    
+    private func didConfigure() async {
```

Comment:
We could move this to an helper function to be reused in all tasks.

Reply:
Done, moved to a helper class.